NPEs occur when building the new invisible projects

After #880 has been merged, I'm seeing NPEs when an invisible project is a parameter of 

https://github.com/eclipse/eclipse.jdt.ls/blob/29d339e5fd96cd74dba740d8cdec5501e5a54d05/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java#L597-L599

turns out `resource.getRawLocationURI()` returns null (as allowed per the API) .

Seen when calling the build command: 
```
[Trace - 7:18:49 PM] Sending request 'java/buildWorkspace - (5)'.
[Trace - 7:18:49 PM] Received notification 'window/logMessage'.
Params: {
    "type": 3,
    "message": "6-Dec-2018 7:18:49 PM >> java/buildWorkspace (incremental)"
}


[Info  - 7:18:49 PM] 6-Dec-2018 7:18:49 PM >> java/buildWorkspace (incremental)
[Trace - 7:18:49 PM] Received response 'java/buildWorkspace - (5)' in 20ms. Request failed: Internal error. (-32603).
Error data: "java.util.concurrent.CompletionException: java.lang.NullPointerException
java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:604)
java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:443)
java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.NullPointerException
org.eclipse.jdt.ls.core.internal.ResourceUtils.fixURI(ResourceUtils.java:172)
 org.eclipse.jdt.ls.core.internal.JDTUtils.getFileURI(JDTUtils.java:598)
org.eclipse.jdt.ls.core.internal.handlers.BuildWorkspaceHandler.buildWorkspace(BuildWorkspaceHandler.java:59)
 org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$24(JDTLanguageServer.java:743)
 org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$26(JDTLanguageServer.java:793)
 java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
... 6 more"

```

I also saw it occur in WorkspaceDiagnosticsHandler, although I'm not able to reproduce it right now

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java
@@ -595,7 +595,7 @@ public final class JDTUtils {
 	 * @return
 	 */
 	public static String getFileURI(IResource resource) {
-		return ResourceUtils.fixURI(resource.getRawLocationURI() != null ? resource.getRawLocationURI() : resource.getLocationURI());
+		return ResourceUtils.fixURI(resource.getRawLocationURI() == null ? resource.getLocationURI() : resource.getRawLocationURI());
 	}
 
 	public static IJavaElement findElementAtSelection(ITypeRoot unit, int line, int column, PreferenceManager preferenceManager, IProgressMonitor monitor) throws JavaModelException {

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java
@@ -595,7 +595,7 @@ public final class JDTUtils {
 	 * @return
 	 */
 	public static String getFileURI(IResource resource) {
-		return ResourceUtils.fixURI(resource.getRawLocationURI());
+		return ResourceUtils.fixURI(resource.getRawLocationURI() != null ? resource.getRawLocationURI() : resource.getLocationURI());
 	}
 
 	public static IJavaElement findElementAtSelection(ITypeRoot unit, int line, int column, PreferenceManager preferenceManager, IProgressMonitor monitor) throws JavaModelException {

